planner/core/find_best_task.go

```diff
@@ -769,6 +777,18 @@ func (is *PhysicalIndexScan) addPushedDownSelection(copTask *copTask, p *DataSou
 	}
 }
 
+// splitSelCondsWithVirtualColumn filter the select conditions which contain virtual column
+func splitSelCondsWithVirtualColumn(conds []expression.Expression) ([]expression.Expression, []expression.Expression) {
+	var filterConds []expression.Expression
+	for i := len(conds) - 1; i >= 0; i-- {
+		if expression.ContainVirtualColumn(conds[i : i+1]) {
+			filterConds = append(filterConds, conds[i])
+			conds = append(conds[:i], conds[i+1:]...)
```

Comment:
Would this have risk of `index out of range`? e.g, if last item of `conds` contains virtual column.

Reply:
I don't think it would, the logic is copy from `eliminate projecyion`, and the `i` here goes from `len(conds) - 1`. Besides, I believe the test have already covered the case "last item of conds contains virtual column."